compiler/rustc_errors/src/json.rs

```diff
@@ -263,6 +289,16 @@ struct ArtifactNotification<'a> {
     emit: &'a str,
 }
 
+#[derive(Serialize)]
+struct SectionTimestamp<'a> {
+    /// Name of the section
+    name: &'a str,
+    /// Start/end of the section
+    kind: &'a str,
+    /// Microseconds elapsed since some predetermined point in time (~start of the rustc process).
```

Comment:
Hm, I wonder if emitting a SystemTime (i.e., unix timestamp -- likely still in microseconds) would perhaps be better in some ways. Placing "time since start" on Cargo's timings graph seems potentially painful -- that really wants an absolute time, IMO, otherwise it seems quite likely we'll quickly hit translation issues (e.g., due to delay in reading this event and needing to re-compute to the time we'd draw on the graph, causing sometimes *negative* time to be seemingly spent in things).

Reply:
Switched to UNIX timestamp in microseconds.